// @mui/joy/modern/ListDivider/ListDivider.js
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import _extends from "@babel/runtime/helpers/esm/extends";
const _excluded = ["component", "className", "children", "inset"];
import * as React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { unstable_capitalize as capitalize } from '@mui/utils';
import composeClasses from '@mui/base/composeClasses';
import { styled, useThemeProps } from '../styles';
import { getListDividerUtilityClass } from './listDividerClasses';
import RowListContext from '../List/RowListContext';
import { jsx as _jsx } from "react/jsx-runtime";

const useUtilityClasses = ownerState => {
  const slots = {
    root: ['root', ownerState.inset && `inset${capitalize(ownerState.inset)}`]
  };
  return composeClasses(slots, getListDividerUtilityClass, {});
};

const ListDividerRoot = styled('li', {
  name: 'JoyListDivider',
  slot: 'Root',
  overridesResolver: (props, styles) => styles.root
})(({
  theme,
  ownerState
}) => _extends({
  border: 'none'
}, ownerState.row && {
  '--List-divider-marginX': 'calc(var(--List-gap) + var(--List-divider-gap))',
  borderInlineStart: '1px solid',
  marginBlock: 0,
  marginInline: 'var(--List-divider-marginX)'
}, !ownerState.row && _extends({}, ownerState['data-first-child'] === undefined && {
  marginBlockStart: 'calc(var(--List-gap) + var(--List-divider-gap))'
}, {
  marginBlockEnd: 'var(--List-divider-gap)',
  marginInline: 'calc(-1 * var(--List-padding))'
}, ownerState.inset === 'gutter' && {
  marginInlineStart: 'var(--List-item-paddingLeft)',
  marginInlineEnd: 'var(--List-item-paddingRight)'
}, ownerState.inset === 'startDecorator' && {
  marginInlineStart: 'var(--List-item-paddingLeft)'
}, ownerState.inset === 'startContent' && {
  marginInlineStart: 'calc(var(--List-item-paddingLeft) + var(--List-decorator-width))'
}, {
  borderBlockEnd: '1px solid'
}), {
  borderColor: theme.vars.palette.divider,
  listStyle: 'none'
}));
const ListDivider = /*#__PURE__*/React.forwardRef(function ListDivider(inProps, ref) {
  const props = useThemeProps({
    props: inProps,
    name: 'JoyListDivider'
  });
  const row = React.useContext(RowListContext);

  const {
    component,
    className,
    children,
    inset
  } = props,
        other = _objectWithoutPropertiesLoose(props, _excluded);

  const ownerState = _extends({
    inset,
    row
  }, props);

  const classes = useUtilityClasses(ownerState);
  return /*#__PURE__*/_jsx(ListDividerRoot, _extends({
    ref: ref,
    as: component,
    className: clsx(classes.root, className),
    ownerState: ownerState,
    role: "separator",
    "aria-orientation": row ? 'horizontal' : 'vertical'
  }, other, {
    children: children
  }));
});
process.env.NODE_ENV !== "production" ? ListDivider.propTypes
/* remove-proptypes */
= {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // |     To update them edit TypeScript types and run "yarn proptypes"  |
  // ----------------------------------------------------------------------

  /**
   * The content of the component.
   */
  children: PropTypes.node,

  /**
   * Override or extend the styles applied to the component.
   */
  classes: PropTypes.object,

  /**
   * @ignore
   */
  className: PropTypes.string,

  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: PropTypes.elementType,

  /**
   * The empty space on the side(s) of the divider.
   * This prop has no effect on the divider if the nearest parent List has `row` prop set to `true`.
   */
  inset: PropTypes
  /* @typescript-to-proptypes-ignore */
  .oneOfType([PropTypes.oneOf(['gutter', 'startDecorator', 'startContent']), PropTypes.string]),

  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])), PropTypes.func, PropTypes.object])
} : void 0;
export default ListDivider;
